### PR TITLE
can filter types of cans

### LINKED ISSUE #29: Make app able to filter can types
We want to be able to display just green waste, just garbage, or just recycling cans on the map

### PR DESCRIPTION
**Pull request by ubelhj**
App can filter different can types

Known issue that switch labels are hard to read in dark mode

Closes #29

### CODE REVIEW

File: FACReact/App/CanMarkers.js [hunk 1]
```diff
@@ -2,16 +2,30 @@ import React from 'react';
...
+        if (!showRecycling) {
+          return;
+        }
         color = 'blue'
       } else if (location.properties.isCompost) {
+        if (!showCompost) {
+          return;
+        }
         color = 'green'
       }
```
Comment (ryantannenberg) [inline FACReact/App/CanMarkers.js]: The logic of the if-statements here is unclear, could be more streamlined

**Review by ryantannenberg — COMMENTED** [review]
Code is missing comments to explain workflow

File: FACReact/App/FACMap.android.js [hunk 2]
```diff
@@ -85,6 +92,10 @@ export class FACMap extends Component {
     this.setState({showGarbage: id === 1, showRecycling: id === 2, showCompost: id === 3});
   }
 
+  toggleFilterModal(visible) {
```
Comment (ubelhj) [inline FACReact/App/FACMap.android.js]: Could have comment to explain usage

File: FACReact/App/FACMap.android.js [hunk 3]
```diff
@@ -145,22 +170,101 @@ export class FACMap extends Component {
...
+                <Button title="cancel" onPress={() => {this.toggleFilterModal(!this.state.filterModalVisible)}}></Button>
+                <Button title="apply" onPress={() => {
+                  this.toggleFilterModal(!this.state.filterModalVisible);
+                  // construct search message to be displayed to user
+                  let newSearchMessage = [
+                    {value: this.state.modalGarbage, text: 'garbage'},
+                    {value: this.state.modalRecycling, text: 'recycling'},
+                    {value: this.state.modalCompost, text: 'compost'}
+                  ].filter(item => item.value).map(item => item.text).join(', ');
+                  this.setState({customSearch: true, searchMessage: newSearchMessage.charAt(0).toUpperCase() + newSearchMessage.slice(1)});
```
Comment (ubelhj) [inline FACReact/App/FACMap.android.js]: Not sure why you'd need to uppercase the first character if the string is just defined above. Could just be 'Garbage'. Might be a misunderstanding of the code.

File: FACReact/App/CanMarkers.js (hunk 1 above)
Comment (ubelhj) [reply FACReact/App/CanMarkers.js]: Ryan has fixed this in commit c90da8d. Thanks Ryan!

File: FACReact/App/FACMap.android.js (hunk 2 above)
Comment (ryantannenberg) [reply FACReact/App/FACMap.android.js]: Fixed in commit 6522e83

File: FACReact/App/FACMap.android.js (hunk 3 above)
Comment (ryantannenberg) [reply FACReact/App/FACMap.android.js]: Fixed in commit f58afc6

**Pull request merged**